compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -257,6 +257,14 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
 
     override def transform(tree: Tree)(using Context): Tree =
       try tree match {
+        case CaseDef(pat, _, _) =>
+          val gadtCtx =
+           pat.getAttachment(typer.Typer.InferredGadtConstraints) match
```

Comment:
If they're not used afterwards we don't want to keep them around:
```suggestion
           pat.removeAttachment(typer.Typer.InferredGadtConstraints) match
```